apps/web/app/routes/app/freeze/_index/_route.tsx

```diff
@@ -84,21 +91,23 @@ export async function action({ request }: ActionFunctionArgs) {
     return json(formError(parsed.errors));
   }
 
+  const securityCouncil = await securityCouncilAddress();
+
   const data = parsed.data;
 
   let nonce: bigint;
   switch (data.type) {
     case "SOFT_FREEZE":
-      nonce = await councilSoftFreezeNonce();
+      nonce = await securityCouncilSoftFreezeNonce(securityCouncil);
```

Comment:
Is there a case where we are going to call `securityCouncilSoftFreezeNonce` with some other thing than that contract? Maybe we can remove that parameter.

Reply:
The same for the other 3.

Reply:
`securityCouncil` is the security council's address

Reply:
Yes, totally. that's why I wonder if we can't just get the address inside the function.